flex_common.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+
+from conans import AutoToolsBuildEnvironment, tools
+from conans.util.env_reader import get_env
+import os
+import shutil
+import tempfile
+
+
+def conanfile_os(conanfile):
```

Comment:
is it possible to make it a base class, not just a set of free functions?

Reply:
I can make it a class, but cannot let the other classes inherit it.